torch/_inductor/compile_fx.py

```diff
@@ -827,11 +841,8 @@ def fx_codegen_and_compile(
                         hasattr(out, "layout")
                         and len(free_unbacked_symbols(out.layout.stride)) == 0
                     ):
-                        output_strides.append(
-                            tuple(
-                                V.graph.sizevars.size_hint(s) for s in out.layout.stride
-                            )
-                        )
+                        # Convert to string for eval on the load path
+                        output_strides.append(tuple(str(s) for s in out.layout.stride))
```

Comment:
It would be much better to use the "real" printer that produce_guards is using, as it ensures that we are consistently printing for Python evaluation. That printer is ShapeGuardPrinter. Note that I do think that at present these coincide relatively closely except for relations (but relations ought not to show up in stride computation), but I could easily imagine refactors where they diverge (in particular, if we decide to clean up SYMPY_INTERP by modifying the printer to go directly to what needs to be bound...)

Reply:
If you are not going to use the correct printer or refactor ShapeGuardPrinter so it is usable in this situation, then there is a silent coupling here and you should try your best to make it obvious that this needs to be updated too if someone does change how the produce guards printing works and modifies SYMPY_INTERP. As it stands, this code is very hard to discover and only would get exercised in obscure places.